storybook CI/CD randomly fails

this error is non-deterministic. Simple re-try/re-run solves the problem. It's a CI-only, not reproducible locally.

e.g. https://github.com/Waog/portfolio/actions/runs/24233029783/attempts/2?pr=127

```
Run # Start static Storybook server in background on port 6006

 NX   Running target static-storybook for project frontend and 1 task it depends on:



> nx run frontend:build-storybook:ci

info => Cleaning outputDir: dist/storybook/frontend
info
 => Loading presets
info
 => Building manager..
info => Manager built (192 ms)
info => Building preview..
WARN No story files found for the specified pattern: apps/frontend/src/app/**/*.@(mdx|stories.@(js|jsx|ts|tsx))
info Addon-docs: using MDX3
info => Using implicit CSS loaders
info => Using angular browser target options from "frontend:build"
info => Using angular project with "tsConfig:/home/runner/work/portfolio/portfolio/apps/frontend/.storybook/tsconfig.json"
Browserslist: browsers data (caniuse-lite) is 10 months old. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
info => Copying static files: apps/frontend/public at dist/storybook/frontend
info => Using default Webpack5 setup
Error: Timed out waiting for: http://localhost:6006
    at /home/runner/work/portfolio/portfolio/node_modules/wait-on/lib/wait-on.js:131:31
    at doInnerSub (/home/runner/work/portfolio/portfolio/node_modules/rxjs/dist/cjs/internal/operators/mergeInternals.js:22:31)
    at outerNext (/home/runner/work/portfolio/portfolio/node_modules/rxjs/dist/cjs/internal/operators/mergeInternals.js:17:70)
    at OperatorSubscriber._this._next (/home/runner/work/portfolio/portfolio/node_modules/rxjs/dist/cjs/internal/operators/OperatorSubscriber.js:33:21)
    at Subscriber.next (/home/runner/work/portfolio/portfolio/node_modules/rxjs/dist/cjs/internal/Subscriber.js:51:18)
    at AsyncAction.work (/home/runner/work/portfolio/portfolio/node_modules/rxjs/dist/cjs/internal/observable/timer.js:28:28)
    at AsyncAction._execute (/home/runner/work/portfolio/portfolio/node_modules/rxjs/dist/cjs/internal/scheduler/AsyncAction.js:79:18)
    at AsyncAction.execute (/home/runner/work/portfolio/portfolio/node_modules/rxjs/dist/cjs/internal/scheduler/AsyncAction.js:67:26)
    at AsyncScheduler.flush (/home/runner/work/portfolio/portfolio/node_modules/rxjs/dist/cjs/internal/scheduler/AsyncScheduler.js:38:33)
    at listOnTimeout (node:internal/timers:581:17)
WARN /home/runner/work/portfolio/portfolio/apps/frontend/.storybook/main.ts is part of the TypeScript compilation but it's unused.
WARN Add only entry points to the 'files' or 'include' properties in your tsconfig.
WARN asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
WARN This can impact web performance.
WARN Assets: 
WARN   9571.527a9f75.iframe.bundle.js (252 KiB)
WARN   8689.71b48679.iframe.bundle.js (736 KiB)
WARN   114.3be6c949.iframe.bundle.js (3.25 MiB)
WARN
   8849.9a20117c.iframe.bundle.js (1.36 MiB)
WARN   5535.7b303aaf.iframe.bundle.js (263 KiB)
WARN   7824.3092234b.iframe.bundle.js (888 KiB)
WARN entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
WARN Entrypoints:
WARN   main (3.29 MiB)
WARN       runtime~main.c71944d7.iframe.bundle.js
WARN       114.3be6c949.iframe.bundle.js
WARN       main.fb0000374ba35c94648e.css
WARN       main.58902ee0.iframe.bundle.js
WARN 
info => Preview built (56 s)
info => Output directory: /home/runner/work/portfolio/portfolio/dist/storybook/frontend

> nx run frontend:static-storybook:ci --port=6006

 NX   Nx Daemon is not enabled. Static server is not watching for changes.
🔁 > nx run frontend:build-storybook:ci  [local cache]
 NX   Successfully ran target build-storybook for project frontend
Nx read the output from the cache instead of running the command for 1 out of 1 tasks.
Starting up http-server, serving dist/storybook/frontend

http-server version: 14.1.1

http-server settings: 
CORS: true
Cache: -1 seconds
Connection Timeout: 120 seconds
Directory Listings: visible
AutoIndex: visible
Serve GZIP Files: false
Serve Brotli Files: false
Default File Extension: none

Available on:
  http://localhost:6006
Unhandled requests will be served from: http://localhost:6006?
Hit CTRL-C to stop the server
Error: Process completed with exit code 1.
```

--- a/.github/workflows/test-storybook.yml
+++ b/.github/workflows/test-storybook.yml
@@ -35,7 +35,7 @@ jobs:
           STORYBOOK_PID=$!
 
           # Wait for Storybook to be ready
-          npx wait-on http://localhost:6006 --timeout 60000
+          npx wait-on http://localhost:6006 --timeout 300000
 
           # Run Storybook tests
           npx nx test-storybook frontend --url=http://localhost:6006

--- a/.github/workflows/test-storybook.yml
+++ b/.github/workflows/test-storybook.yml
@@ -25,8 +25,19 @@ jobs:
       - name: 🎭 Setup Playwright environment
         uses: ./.github/actions/setup-playwright
 
+      - name: 💾 Cache wait-on for Storybook tests
+        id: wait-on-cache
+        uses: actions/cache@v4
+        with:
+          path: .wait-on
+          key: ${{ runner.os }}-storybook-wait-on-${{
+            hashFiles('.github/workflows/test-storybook.yml') }}
+          restore-keys: |
+            ${{ runner.os }}-storybook-wait-on-
+
       - name: 📚 Install wait-on for Storybook tests
-        run: npm install --no-save wait-on
+        if: steps.wait-on-cache.outputs.cache-hit != 'true'
+        run: npm install --no-save --prefix .wait-on wait-on
 
       - name: 📖 Test Storybook
         run: |
@@ -35,7 +46,7 @@ jobs:
           STORYBOOK_PID=$!
 
           # Wait for Storybook to be ready
-          npx wait-on http://localhost:6006 --timeout 300000
+          .wait-on/node_modules/.bin/wait-on http://localhost:6006 --timeout 300000
 
           # Run Storybook tests
           npx nx test-storybook frontend --url=http://localhost:6006

